### ISSUE TITLE
Task groups

### ISSUE THREAD

**Issue by Yohe-Am** [metatypedev/ghjk#71]
```ts
// NOTE: whitespace in name
task("build front", async () => { /* build frontend */ } )
task("build back", async () => { /* build backend */ })
```
 
It'll then be accessed by...

```shell
$ ghjk x build front
$ ghjk x build back
```

- If only `ghjk x build`, is invoked, list the available tasks.
- On `ghjk x build --run`, run all tasks under it in parallel.
- If another task is explicitly defined under the name `build`, it overrides this group header task.

_Originally posted by @Yohe-Am in https://github.com/metatypedev/ghjk/issues/18#issuecomment-1933003330_
